include/inviwo/core/ports/imageport.h

```diff
@@ -389,6 +393,20 @@ Document BaseImageInport<N>::getInfo() const {
     return doc;
 }
 
+template <size_t N>
+bool BaseImageInport<N>::hasData() const {
+    if constexpr (N == 0) {
+        return this->isConnected() && this->begin() != this->end();
```

Comment:
What about this case?

Reply:
I kept this the same as for DataInport. However, I agree that the logic here seems flawed. Should be 
`isConnected() && util::any_of…`